platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/FilterInjector.java

```diff
@@ -0,0 +1,202 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.pax.web.jetty;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import javax.servlet.DispatcherType;
+import javax.servlet.Filter;
+import javax.servlet.FilterRegistration;
+import javax.servlet.ServletContext;
+import javax.servlet.SessionCookieConfig;
+import javax.servlet.http.HttpSessionListener;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Injects the {@link SecurityJavaSubjectFilter} into the {@link ServletContext} with properties
+ * filterName = {@link FilterInjector#SECURITY_JAVA_SUBJECT_FILTER}, urlPatterns = {@link
+ * FilterInjector#ALL_URLS}, and no servlet name as the {@link ServletContext} is being created.
+ * Sets up all {@link javax.servlet.Servlet}s with the {@link SecurityJavaSubjectFilter} in the
+ * {@link javax.servlet.FilterChain}.
+ */
+public class FilterInjector implements EventListenerHook {
```

Comment:
❓ I'm curious how this change makes things more stable, since we're still relying on the FilterInjector? Is it just because the FilterInjector is more tightly coupled to the Jetty server startup now?

Reply:
So the only thing that the FilterInjector is used for now is to run attach java security subject to the thread. Everything else is happening within Jetty, the security filters and the shiro subject attachment. @stustison said that the java subject would be used for an action such as feature install via admin console. So this is necessary but shouldn't cause problems that we currently see. (Jetty didn't allow us to attach java security subject based on how it was designed)